### PR TITLE
Refactor api and add v2 API (beta)

### PR DESCRIPTION
**Pull request by kindermoumoute**
Summary of changes:
- @kjlyon's work on API tests.
- [x] Add the API interface
- [x] Move v1 under mgmt/rest/v1
- [x] Implements v2

@intelsdi-x/snap-maintainers

### CODE REVIEW

**Comment by kindermoumoute** [comment]
The review of this PR has been done in #1474, this PR is just a rebase in order to have clear commits in the log.

File: mgmt/rest/v2/error.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import (
+	"fmt"
+
+	"errors"
+
+	"github.com/intelsdi-x/snap/core/serror"
+)
+
+const (
+	ErrPluginAlreadyLoaded     = "plugin is already loaded"
```
Comment (jcooklin) [inline mgmt/rest/v2/error.go]: Let's take this opportunity to make the errors consistent in terms of language, punctuation and capitalization.

File: mgmt/rest/v2/error.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	ErrTaskNotFound            = "Task not found"
+	ErrTaskDisabledNotRunnable = "Task is disabled. Cannot be started"
```
Comment (jcooklin) [inline mgmt/rest/v2/error.go]: "Cannot be started" can probably be dropped.

File: mgmt/rest/v2/error.go (hunk 1 above)
Comment (kindermoumoute) [inline mgmt/rest/v2/error.go]: Done.

File: mgmt/rest/v2/error.go (hunk 2 above)
Comment (kindermoumoute) [inline mgmt/rest/v2/error.go]: Done.

**Comment by jcooklin** [comment]
@IRCody, @nanliu, @candysmurf can we get another LGTM?

**Comment by candysmurf** [comment]
LGTM 👍

**Pull request merged**